src/ayab/encoders.cpp

```diff
@@ -136,15 +136,32 @@ Carriage_t Encoders::detectCarriageLeft() {
 }
 
 Carriage_t Encoders::detectCarriageRight() {
-  uint16_t hallValue = analogRead(EOL_PIN_R);
   if (m_machineType == MachineType::Kh910) {
-    // Due to an error in wiring on the shield, the KH910/KH950 right sensor
-    // only triggers for the K carriage, and with a low voltage so we can't 
-    // use the same logic as for other machines.
-    if (hallValue < FILTER_R_MIN[static_cast<uint8_t>(m_machineType)]) {
+    pinMode(EOL_PIN_R_L, INPUT_PULLUP);
+    // Set EOL_PIN_R_DETECT to LOW to detect if it is connected to EOL_PIN_R_L
+    pinMode(EOL_PIN_R_DETECT, OUTPUT);
```

Comment:
I would move pinMode after digitalWrite (make sure that pin is low when output is enabled and avoid a transient short-circuit on a fixed shield/interface if output was HIGH for whatever reason and machine drive EOL_PIN_R low)